File: net/QACover/Giis.Qacover.Portable/Jdk14StringBuilder.cs
using System.Text;


namespace Giis.Qacover.Portable
{
	/// <summary>
	/// Utilizar esta clase cuando se va a hacer downgrade de codigo a jdk 1.4, ya que no tiene StringBuilder:
	/// El codigo con el downgrade implmentara otra version de esta clase.
	/// </summary>
	public class Jdk14StringBuilder
	{
		internal StringBuilder s = new StringBuilder();

		public virtual int Length()
		{
			return s.Length;
		}

		public virtual void Append(string str)
		{
			s.Append(str);
		}

		public override string ToString()
		{
			return s.ToString();
		}
	}
}

File: qacover-model/src/main/java/giis/qacover/portable/Jdk14StringBuilder.java
package giis.qacover.portable;

/**
 * Utilizar esta clase cuando se va a hacer downgrade de codigo a jdk 1.4, ya
 * que no tiene StringBuilder: El codigo con el downgrade implmentara otra
 * version de esta clase.
 */
public class Jdk14StringBuilder {
	StringBuilder s = new StringBuilder();

	public int length() {
		return s.length();
	}

	public void append(String str) {
		s.append(str);
	}

	@Override
	public String toString() {
		return s.toString();
	}
}
